### PR TITLE
Add REST API for purge audit log

### PR DESCRIPTION
**Pull request by stonezdj**
Add rest API for purge audit log
  Extract scheduler, execution, task from GC controller for common use

Signed-off-by: stonezdj <stonezdj@gmail.com>

Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #(issue)

Please indicate you've done the following:
- [ ] Well Written Title and Summary of the PR
- [ ] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [ ] Accepted the DCO. Commits without the DCO will delay acceptance.
- [ ] Made sure tests are passing and test coverage is added if needed.
- [ ] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

File: src/controller/jobservice/execution.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
+//  Copyright Project Harbor Authors
```
Comment (wy65701436) [inline src/controller/jobservice/execution.go:1]: duplication with /src/task/execution_controller.go

File: api/v2.0/swagger.yaml [hunk 2]
```diff
@@ -4174,6 +4174,170 @@ paths:
           $ref: '#/responses/403'
         '500':
           $ref: '#/responses/500'
+  /system/purge:
```
Comment (wy65701436) [inline api/v2.0/swagger.yaml:4177]: Will we expand the purge to support other kinds of resources? If not, we can use /system/purgeAudit?

File: src/controller/jobservice/task.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import (
+	"context"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/lib/q"
+	"github.com/goharbor/harbor/src/pkg/task"
+)
+
+var (
+	// TaskCtl ...
+	TaskCtl = NewTaskController()
```
Comment (chlins) [inline src/controller/jobservice/task.go:26]: What's the difference between this controller and src/controller/task.Ctl?

File: api/v2.0/swagger.yaml (hunk 2 above)
Comment (stonezdj) [reply api/v2.0/swagger.yaml]: Updated

File: src/server/v2.0/handler/purge.go [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+		policy.DryRun = dryRun
+	}
+	if includeOperations, ok := parameters[common.PurgeAuditIncludeOperations].(string); ok {
+		policy.IncludeOperations = includeOperations
+	}
+	if retentionHour, ok := parameters[common.PurgeAuditRetentionHour]; ok {
+		if rh, ok := retentionHour.(json.Number); ok {
+			ret, err := rh.Int64()
+			if err != nil {
+				log.Warningf("failed to convert audit_retention_hour, error: %v", err)
```
Comment (chlins) [inline src/server/v2.0/handler/purge.go:108]: no need return error?

File: src/server/v2.0/handler/purge.go [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+
+func (p *purgeAPI) updateSchedule(ctx context.Context, vendorType, cronType, cron string, policy pg.JobPolicy, extraParams map[string]interface{}) error {
+	if err := p.schedulerCtl.Delete(ctx, vendorType); err != nil {
+		return err
+	}
+	return p.createSchedule(ctx, vendorType, cronType, cron, policy, extraParams)
+}
+
+func (p *purgeAPI) GetPurgeHistory(ctx context.Context, params purge.GetPurgeHistoryParams) middleware.Responder {
+	if err := p.RequireSystemAccess(ctx, rbac.ActionRead, rbac.ResourcePurgeAuditLog); err != nil {
```
Comment (chlins) [inline src/server/v2.0/handler/purge.go:134]: This function aims to list purge histories, so suggest use `rbac.ActionList`.

File: src/server/v2.0/handler/purge.go [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+		Kind:       exec.Trigger,
+		Parameters: string(extraAttrsString),
+		Status:     exec.Status,
+		Schedule: &model.ScheduleParam{
+			Type: exec.Trigger,
+		},
+		CreationTime: exec.StartTime,
+		UpdateTime:   exec.EndTime,
+	}
+	res.ToSwagger()
```
Comment (chlins) [inline src/server/v2.0/handler/purge.go:208]: useless?

Comment (stonezdj) [reply src/server/v2.0/handler/purge.go:208]: Updated

File: src/server/v2.0/handler/purge.go (hunk 4 above)
Comment (stonezdj) [reply src/server/v2.0/handler/purge.go:108]: updated

**Review by chlins — APPROVED** [review]
lgtm

**Review by wy65701436 — APPROVED** [review]
lgtm

**Pull request merged**